models/repo/repo_list.go

```diff
@@ -730,3 +730,21 @@ func GetUserRepositories(opts *SearchRepoOptions) (RepositoryList, int64, error)
 	repos := make(RepositoryList, 0, opts.PageSize)
 	return repos, count, db.SetSessionPagination(sess, opts).Find(&repos)
 }
+
+func GetPrimaryRepoLanguageList(ctx context.Context) (LanguageStatList, error) {
+	languageList := make(LanguageStatList, 0)
+
+	err := db.GetEngine(ctx).
+		Table("language_stat").
+		Cols("language").
+		Where(builder.Eq{"is_primary": true}).
+		Distinct("language").
+		OrderBy("language").
+		Find(&languageList)
+	if err != nil {
+		return nil, err
+	}
+
+	languageList.LoadAttributes()
+	return languageList, nil
+}
```

Comment:
```suggestion

func GetPrimaryRepoLanguageList(ctx context.Context) ([]string, error) {
	languageList := make([]string, 0)
  err := db.GetEngine(ctx).
		Table("language_stat").
		Cols("language").
		Where(builder.Eq{"is_primary": true}).
		Distinct("language").
		OrderBy("language").
		Find(&languageList)
	return languageList, err
}
```

Reply:
We also need the Color, which we get with `LoadAttributes()`

Reply:
Would you like me to get rid of the color icons or maybe return a `map[string]string` with only the language and the color

Reply:
Color icons are a nice touch, let's keep them.

Reply:
> We also need the Color, which we get with `LoadAttributes()`

If ids haven't been loaded from database, how does the `LoadAttributes` work?
And since there is `Distinct`, not all databases support loading ids. Except you add `id` as another `Distinct` columns.

Reply:
This is the current Code of `LoadAttributes()`:
```go
// LoadAttributes loads attributes
func (stats LanguageStatList) LoadAttributes() {
	for i := range stats {
		stats[i].Color = enry.GetColor(stats[i].Language)
	}
}
```
As you can see, it only uses the Language Column and not the ID Column. Using the ID Column for the Color would not make much sense, as this is different on every Database.